backend/src/test/java/ee/tlu/forum/controller/ThreadControllerTest.java

```diff
@@ -0,0 +1,266 @@
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+
+        AddNewThreadInput form = new AddNewThreadInput();
+        form.setUsername("user1");
+        form.setTitle("title");
+        form.setContent("content");
+        String formJson = new ObjectMapper().writeValueAsString(form);
+
+        when (threadService.createThread(any(AddNewThreadInput.class)))
+                .thenReturn(t1);
+
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.post("/api/thread/add")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(formJson);
+
+        // when then
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().is(201))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title", Matchers.is("title")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.text", Matchers.is("content")));
+    }
+
+    @Test
+    @DisplayName("Returns edited thread - PATCH /api/thread/edit")
+    void editThread() throws Exception {
+        // given
+        User user = new User(1L,
+                "user1",
+                "test1@test.com",
+                "aaa",
+                new ArrayList<>(),
+                "",
+                "",
+                new ArrayList<>(),
+                new ArrayList<>());
+        Thread t1 = new Thread(1L, "content", "title", user, new ArrayList<>());
+
+        Map<String, String> form = new HashMap<>();
+        form.put("title", t1.getTitle());
+        String formJson = new ObjectMapper().writeValueAsString(form);
+
+        when (threadService.editThread(any()))
+                .thenReturn(t1);
+
+        RequestBuilder requestBuilder = MockMvcRequestBuilders.patch("/api/thread/edit")
+                .accept(MediaType.APPLICATION_JSON)
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(formJson);
+
+        // when then
+        mockMvc.perform(requestBuilder)
+                .andExpect(MockMvcResultMatchers.status().is(200))
+                .andExpect(MockMvcResultMatchers.content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.id", Matchers.is(1)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.title", Matchers.is("title")))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.text", Matchers.is("content")));
+    }
+
+
+    @Test
+    @DisplayName("Returns all posts of a thread edited thread - GET /api/thread/{id}/posts")
```

Comment:
I wonder whether we should test the functions that return 'all' or 'a list' with multiples? In  this case, more than one post?

Reply:
Sure - added to todo